### DISCUSSION TITLE
regarding the mismatch between the alarm information and the rule directory

### DISCUSSION THREAD

**Discussion by 1998729** [jertel/elastalert2#1064] [Q&A] [answered]
Hello everyone, I am currently using elastalert2, the image version is 2.3.0, and the es version is 6.8. I have found a phenomenon that the alarm content and query conditions do not match those defined in my rule directory, which is very strange.

**Comment by 1998729** [comment]
At present, there is a synchronous job function, which will regularly get the rule information from our database, and then write it into the rules specified by elastalert2 in batches, in the form of a file, ending with .yaml

**Comment by jertel** [comment]
The log file will tell you if it is detecting that the rule alarm content and filters have changed. You should see a log line such as `Reloading configuration for rule "name"`. Are you seeing this?

**Reply by 1998729** [reply]
> The log file will tell you if it is detecting that the rule alarm content and filters have changed. You should see a log line such as `Reloading configuration for rule "name"`. Are you seeing this?

Thanks to the author's reply, I researched it and found that the values concatenated with query_string are uppercase, but the quotation marks are not added to match other query conditions.

**Reply by 1998729** [reply]
wrong query method:

filter:
- query:
    query_string:
      query: 'business: open AND http_host: AIKNLLL'

correct query:
 filter:
- query:
    query_string:
      query: 'business: open AND http_host: "AIKNLLL"'

**Comment by 1998729** [comment, marked as answer]
Problem solved, cause was using es pose issue.
